Data/Csv.hs

```diff
@@ -124,6 +124,50 @@ import Data.Csv.Types
 --
 -- In practice, the return type of 'decode' rarely needs to be given,
 -- as it can often be inferred from the context.
+-- 
+-- Demonstration of reading from a CSV file/ writing to a CSV file
```

Comment:
Perhaps this should be in the incremental API module? You could add a forward reference here (e.g. "For an example of achieving the same using the incremental API, see the 'Data.Csv.Incremental' module.")